fearless/Modules/Staking/Analytics/Common/AnalyticsViewModelFactoryBase.swift

```diff
@@ -0,0 +1,205 @@
+import BigInt
+import SoraFoundation
+
+protocol AnalyticsViewModelItem: Dated {
+    var timestamp: Int64 { get }
+    var amount: BigUInt { get }
+}
+
+class AnalyticsViewModelFactoryBase<T: AnalyticsViewModelItem> {
+    let chain: Chain
+    let balanceViewModelFactory: BalanceViewModelFactoryProtocol
+
+    init(
+        chain: Chain,
+        balanceViewModelFactory: BalanceViewModelFactoryProtocol
+    ) {
+        self.chain = chain
+        self.balanceViewModelFactory = balanceViewModelFactory
+    }
+
+    func createViewModel(
+        from data: [T],
+        priceData: PriceData?,
+        period: AnalyticsPeriod,
+        periodDelta: Int
+    ) -> LocalizableResource<AnalyticsRewardsViewModel> {
+        LocalizableResource { [self] locale in
+            let timestampInterval = period.timestampInterval(periodDelta: periodDelta)
```

Comment:
We can have a case when base date changes for example while app was in background. So it is possible to jump over some values. Consider to store the date which is used for relativistic calculations in presenter.